src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -764,8 +767,8 @@ public bool Remove(TKey key)
             int collisionCount = 0;
             if (buckets != null)
             {
-                int hashCode = (_comparer?.GetHashCode(key) ?? key.GetHashCode()) & 0x7FFFFFFF;
-                int bucket = hashCode % buckets.Length;
+                uint hashCode = (uint)(_comparer?.GetHashCode(key) ?? key.GetHashCode());
+                int bucket = (int)(hashCode % (uint)buckets.Length);
```

Comment:
Some other places avoid the `int` cast by using `uint`, e.g.: `buckets[hashCode % (uint)buckets.Length]`. Why not do the same here?

Reply:
I don't recall if there is a reason related to IL generation or not, I'll double check and remove in case!

Reply:
I misremembered simply it doesn't compile, on array is implicit https://sharplab.io/#v2:EYLgHgbALANALiAlgGwD4AEAMACdBGAbgFgAodAZlwCZsARRAY2wG9Tt3s2PEA7OAbQC62APrAArgwDWAUzgBnbAF5sPccmTESHTto4VsvONgCyeABTij2ABYBDeTYDCAewAmMgJQsdOrr9wAdnZze0dXD2wAUmxLI08xSVkFADoAGRkeAHM4G08CAPZ/bABfYuKDaxMqOL5bB2d3LxZinXRgxOk5eX4wxsiY2rgEiS7UjOzcwS0dMpISoA=

Reply:
What I meant is change `int bucket = ...` to `uint bucket = ...` here and delete the cast.

Reply:
aahhhh ok sorry you're right

Reply:
done